yellow-paper/docs/public-vm/avm-circuit.md

```diff
@@ -51,3 +72,62 @@ An instruction like `ADDRESS` serves as great example because it performs a read
     - memory write
     - flags: `callPointer`, `userMemory = 1` (user memory access)
     - offset: `dstOffset`
+
+## Circuit I/O
+
+### How do "Public Inputs" work in the AVM circuit?
+ZK circuit proof systems generally define some mechanism for "public inputs" for which witness values must be communicated in full to a verifier. The AVM proof system defines its own mechanism for public inputs in which it flags certain trace columns as "public input columns". Any public input columns must be communicated in full to a verifier.
+
+### AVM public inputs structure
+The VM circuit's I/O is defined as the `AvmPublicInputs` structure detailed below:
+```
+AvmSideEffects {
+    newNoteHashes,
+    newNullifiers,
+    newL2ToL1Messages,
+    unencryptedLogs,
+}
+AvmGasResults {
+    l1GasLeft,
+    l2GasLeft,
+    daGasLeft,
+}
+AvmPublicInputs {
+    initialEnvironment: ExecutionEnvironment & {l1GasLeft, l2GasLeft, daGasLeft},
+    calldata: [],
+    sideEffects: AvmSideEffects,
+    storageAccesses,
+    gasResults: {l1GasLeft, l2GasLeft, daGasLeft},
```

Comment:
Why not using the data type AvmGasResults? This was done this way for sideEffects field.

Reply:
I meant to get rid of that struct. Thought it simple enough to just show inline in curly braces `{...}`. Thoughts?